### ISSUE TITLE
flutter_tester --non-interactive should never call beginFrame/drawFrame

### ISSUE THREAD

**Issue by FlutterIssues** [flutter/flutter#178]
<a href="https://github.com/Hixie"><img src="https://avatars.githubusercontent.com/u/551196?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [Hixie](https://github.com/Hixie)**
_Friday Oct 02, 2015 at 18:21 GMT_
_Originally opened as https://github.com/flutter/engine/issues/1457_

---

Running the tests directly, as opposed to through the test harness, you get exceptions because the animation code is getting called from beginFrame() which is being called from the engine directly with a different clock time than the fake clock the test is trying to run the system with.

**Comment by Hixie** [comment]
I've temporarily worked around this in some of the tests by disconnecting the binding's hook into the engine, but that's not a good long-term solution.

**Comment by Hixie** [comment]
@chinmaygarde This bug may be of interest to you. Happy to chat about it if you want.

**Comment by chinmaygarde** [comment]
`BeginFrame` should only be called if a frame is actually requested. If not, this is likely a bug in the engine. I will can verify it tomorrow.

**Comment by Hixie** [comment]
Oh, cool. I may be able to simplify the test harness then. (This bug is old, it may have changed since I filed it.)

**Comment by Hixie** [comment]
@chinmaygarde Looks like the engine is still canning window.onBeginFrame during headless `flutter test`.

To check, I edited `packages/flutter_test/lib/src/binding.dart`, in the `initInstances` method of the `AutomatedTestWidgetsFlutterBinding` class, I changed the last line to:
```dart
    ui.window.onBeginFrame = (Duration timestamp) { print('engine called onBeginFrame!'); };
```

Then I ran `flutter test` in the `flutter_test` directory, and observed that it calls that print statement. By checking the stack trace that that closure gets called with (using `debugPrintStack`) I verified that it is coming from calls to `_beginFrame`.
